hypothesis-python/docs/reference/api.rst

```diff
@@ -3,81 +3,11 @@ API Reference
 
 Reference for non-strategy objects that are part of the Hypothesis API. For documentation on strategies, see the :doc:`strategies reference </reference/strategies>`.
 
-|@given|
---------
+``@given``
+----------
 
 .. autofunction:: hypothesis.given
-
-.. data:: hypothesis.infer
-
-Arguments to |@given|
-~~~~~~~~~~~~~~~~~~~~~
-
-The |@given| decorator may be used to specify which arguments of a function should be parametrized over. You can use either positional or keyword arguments, but not a mixture of both.
-
-For example, all of the following are valid uses:
-
-.. code:: python
-
-  @given(integers(), integers())
-  def a(x, y):
-      pass
-
-  @given(integers())
-  def b(x, y):
-      pass
-
-  @given(y=integers())
-  def c(x, y):
-      pass
-
-  @given(x=integers())
-  def d(x, y):
-      pass
-
-  @given(x=integers(), y=integers())
-  def e(x, **kwargs):
-      pass
-
-  @given(x=integers(), y=integers())
-  def f(x, *args, **kwargs):
-      pass
-
-  class SomeTest(TestCase):
-      @given(integers())
-      def test_a_thing(self, x):
-          pass
-
-The following are not:
-
-.. code:: python
-
-  @given(integers(), integers(), integers())
-  def g(x, y):
-      pass
-
-  @given(integers())
-  def h(x, *args):
-      pass
-
-  @given(integers(), x=integers())
-  def i(x, y):
-      pass
-
-  @given()
-  def j(x, y):
-      pass
-
-The rules for determining what are valid uses of ``given`` are as follows:
-
-1. You may pass any keyword argument to ``given``.
-2. Positional arguments to ``given`` are equivalent to the rightmost named arguments for the test function.
-3. Positional arguments may not be used if the underlying test function has ``*args``, ``**kwargs``, or keyword-only arguments.
-4. Functions tested with ``given`` may not have any defaults.
-
-The reason for the "rightmost named arguments" behaviour is so that using |@given| with instance methods works: ``self`` will be passed to the function as normal and not be parametrized over.
-
-The function returned by given has all the same arguments as the original test, minus those that are filled in by |@given|. Check :ref:`the notes on framework compatibility <framework-compatibility>` to see how this affects other testing libraries you may be using.
+.. autodata:: hypothesis.infer
```

Comment:
needs a sentence or two of exposition; something like "An alias for `...`, the :class:Ellipsis type, which you can pass this to `@given()` or `st.builds()` to indicate that a strategy for that parameter should be inferred from type annotations."